7.2.turtle_sent.py

```diff
@@ -0,0 +1,84 @@
+# Omer Gertler
+
+
+class PostOffice:
+    """A Post Office class. Allows users to message each other.
+
+    :ivar int message_id: Incremental id of the last message sent.
+    :ivar dict boxes: Users' inboxes.
+
+    :param list usernames: Users for which we should create PO Boxes.
+    """
+
+    def __init__(self, usernames):
+        self.message_id = 0
+        self.boxes = {user: [] for user in usernames}
+
+    def send_message(self, sender, recipient, message_title, message_body, urgent=False):
+        """Send a message to a recipient.
+
+        Note: I added 'read' key to the message_details dictionary to enable marking messages
+                as read or not. This key is useful in the "read_inbox" method.
+                The default 'read' value for a new message is False.
+                In addition, I added 'title' to the details to enable
+                searching in the title (look at the exercise requirements). (Omer Gertler).
+        :param str sender: The message sender's username.
+        :param str recipient: The message recipient's username.
+        :param str message_title: The title of the message.
+        :param str message_body: The body of the message.
+        :param urgent: The urgency of the message.
+        :type urgent: bool, optional
+        :return: The message ID, auto incremented number.
+        :rtype: int
+        :raises KeyError: if the recipient does not exist.
+        """
+        user_box = self.boxes[recipient]
+        self.message_id = self.message_id + 1
+        message_details = {
+            'id': self.message_id,
+            'title': message_title,
+            'body': message_body,
+            'sender': sender,
+            'read': False
+        }
+        if urgent:
+            user_box.insert(0, message_details)
+        else:
+            user_box.append(message_details)
+        return self.message_id
+
+    def read_inbox(self, username, n=None):
```

Comment:
'n' should be replaced with a more explanatory name.
Variable names - https://workat.tech/machine-coding/tutorial/writing-meaningful-variable-names-clean-code-za4m83tiesy0 - variable names, abbreviation reference is missing